src/Voron/Data/Tables/TableSchema.cs

```diff
@@ -225,38 +187,65 @@ public void Validate(SchemaIndexDef actual)
                         $"Expected index to have Name='{Name}', got Name='{actual.Name}' instead",
                         nameof(actual));
 
-                if (Type != actual.Type)
+                if (IsGlobal != actual.IsGlobal)
                     throw new ArgumentException(
-                        $"Expected index {Name} to have Type='{Type}', got Type='{actual.Type}' instead",
+                        $"Expected index {Name} to have IsGlobal='{IsGlobal}', got IsGlobal='{actual.IsGlobal}' instead",
                         nameof(actual));
 
-                if (StartIndex != actual.StartIndex)
+                if (Type != actual.Type || actual is not StaticBTreeIndexDef staticBTreeIndexDef)
```

Comment:
Check my above comment about" `<TSelf> where TSelf : AbstractTableIndexDef<TSelf>` construction ". If you decide to go this way then you won't need to cast it here.

Reply:
decided not to go with `TSelf`, so that `TableSchema` can have `_indexes` dictionary with values of type `AbstractTreeIndexDef` (without type argument)